tests/cxx/badinc.cc

```diff
@@ -307,27 +307,22 @@ template<> struct Cc_TemplateStruct<I1_Class, I12> { I1_Class* i; };
 // IWYU: I1_Enum is...*badinc-i1.h
 template<class T = I1_Class, I1_Enum E = I11> class Cc_DeclareOnlyTemplateClass;
 
-// I2_Class has a non-explicit constructor (actually, two), so we need
```

Comment:
This comment might actually help clarify what `No autocast assurance for function definitions` refers to. Consider reading the test after your change; there's no indication that `I2_Class` has implicit constructors and might trigger `autocast assurance`.

I suggest:
```
  // I2_Class has non-explicit constructors, so would normally trigger special
  // rules for autocast. But these rules do not apply to function definitions,
  // so uses of I2_Class below should behave just like I1_Class.
```